basic/splunk/src/test/java/org/springframework/integration/samples/splunk/SplunkWatcher.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.samples.splunk;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Assume;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.springframework.integration.splunk.support.SplunkServer;
+import org.springframework.integration.splunk.support.SplunkServiceFactory;
+
+/**
+ * @author filippo balicchia
+ * @since 4.2
+ */
+public class SplunkWatcher extends TestWatcher {
+
+	private static final Log logger = LogFactory.getLog(SplunkWatcher.class);
+
+	@Override
+	public Statement apply(Statement base, Description description) {
+		try {
+
+			SplunkServer splunkServer = new SplunkServer();
```

Comment:
What is bad here that we rely only on the default port for the Splunk and don't care about that one in the `application.yml`.
I'm not sure that we will be able to deal with the last one, but, at least, I see simple workaround like this:
```
System.setProperty("splunk.port", "" + splunkServer.getPort());
```
That definitely overrides the value in the `application.yml`. Therefore our test is able to pass independently of the preference in case of normal application run.
In addition I added this as well:
```
private final int port;

public SplunkWatcher() {
	this(Service.DEFAULT_PORT);
}

public SplunkWatcher(int port) {
	this.port = port;
}
```
So, I can fix it on merge, but I need to know your opinion regarding capital letters in your name :smile: